### PR TITLE
Cache last scan results

### PR DESCRIPTION
**Pull request by WilsonZiweiWang**
Save the last scan results in the workspace state to reuse it and skip the scan in the next startup. 

Ticket: 14207

### CODE REVIEW

File: client/src/extension.ts
Comment (idillon-sfl) [inline client/src/extension.ts:223]: That's cool not to have the scan starting when we already have the previous version. Things get ready much faster.

However, I wonder, should we start the scan nevertheless in case some changes happened outside of VS Code? (would there be any benefit in keeping a cache if we do so?)

File: client/src/extension.ts
Comment (deribaucourt) [inline client/src/extension.ts:228]: I think we should also write the scan result in bitBakeProjectScanner.scanResult. It's used in several places, notably the selectRecipe function from the commands.

File: client/src/extension.ts
Comment (deribaucourt) [reply client/src/extension.ts:223]: I think the tradeoff is good. Rarely will users modify the project outside VSCode. They can easily start a scan manually through the buttons in that case. We could explain this in the Troubleshooting.md tradeoffs.

**Review by deribaucourt — CHANGES_REQUESTED** [review]

File: client/src/extension.ts
Comment (WilsonZiweiWang) [reply client/src/extension.ts:229]: done. check the new commit

File: client/src/extension.ts
Comment (WilsonZiweiWang) [reply client/src/extension.ts:223]: We could have a setting that lets users control this behaviour to cache the result or not. @deribaucourt What do you think about the changes that happened outside the vscode?

Comment (deribaucourt) [reply client/src/extension.ts:223]: My comment on this disappeared! I'm fine with not handling changes outside of bitbake in any way. Rarely will users have them. If they do, they can easily launch a scan manually form the UI. We can document this in troubleshooting.md if you're really worried IMO.

**Review by deribaucourt — DISMISSED** [review]

File: client/src/extension.ts
Comment (idillon-sfl) [reply client/src/extension.ts:223]: sounds good to me

**Review by idillon-sfl — APPROVED** [review]

**Comment by WilsonZiweiWang** [comment]
I will merge this after the CI fix is merged.

**Review by deribaucourt — APPROVED** [review]

**Pull request merged**
